### PR TITLE
gha: add GOPATH env var to the ppc64le k8s workflow

### PR DESCRIPTION
**Pull request by Amulyam24**
The filtering of testing cases installs/uses yq and expects GOPATH to be present. Hence, add it to the workflow.

Fixes: #9018

### CODE REVIEW

File: .github/workflows/run-k8s-tests-on-ppc64le.yaml [hunk 1]
```diff
@@ -37,6 +37,7 @@ jobs:
       DOCKER_REPO: ${{ inputs.repo }}
       DOCKER_TAG: ${{ inputs.tag }}
       PR_NUMBER: ${{ inputs.pr-number }}
+      GOPATH: ${{ github.workspace }}
```
Comment (fidencio) [inline .github/workflows/run-k8s-tests-on-ppc64le.yaml:40]: out of curiosity, why?
do we build something as part of this workflow?

Comment (Amulyam24) [reply .github/workflows/run-k8s-tests-on-ppc64le.yaml:40]: @fidencio, check this out - https://github.com/kata-containers/kata-containers/blob/main/tests/integration/kubernetes/filter_k8s_test.sh#L25-L31

This is mainly used on the non x86 arches to build/use `yq` which is used to filter out certain test cases.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @Amulyam24!

**Pull request merged**